### PR TITLE
Completion gets in the way attempting to line up arguments

### PR DESCRIPTION
**Pull request by ivanbasov**
Fixes https://github.com/dotnet/roslyn/issues/34237

### CODE REVIEW

**Comment by ivanbasov** [comment]
@dpoeschl , @sharwell please review

File: src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs [hunk 1]
```diff
@@ -63,6 +63,12 @@ internal CompletionSource(ITextView textView, IThreadingContext threadingContext
             // We would like to be sure that nobody changes buffers at the same time.
             AssertIsForeground();
 
+            if (_textView.Selection.Mode == TextSelectionMode.Box)
```
Comment (sharwell) [inline src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs]: 💭 It seems like this should be a separate issue. Do we have tests for this?

Comment (ivanbasov) [reply src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs]: @sharwell, the second fix was a caused by a test failure. This test:
https://github.com/dotnet/roslyn/blob/43368ee4df7f26b693c68d339500f3f1e15427bf/src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb#L2925
worked occasionally due to missing of those two checks.
When I restored one of conditions, it started failing. It requires to restore
`if (_textView.Selection.Mode == TextSelectionMode.Box)`
from the old completion.

**Comment by ivanbasov** [comment]
@dpoeschl , @sharwell please review

File: src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs [hunk 2]
```diff
@@ -109,6 +115,12 @@ internal CompletionSource(ITextView textView, IThreadingContext threadingContext
                 return true;
             }
 
+            // Enter does not trigger completion.
+            if (trigger.Reason == AsyncCompletionData.CompletionTriggerReason.Insertion && trigger.Character == '\n')
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs]: Does this break any Roslyn extenders?

Comment (sharwell) [reply src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs]: Do we ever trigger completion on enter in legacy completion?

Comment (ivanbasov) [reply src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs]: The legacy completion does not trigger completion on Enter:
https://github.com/dotnet/roslyn/blob/bd1e0c6cb23e552d390c1b3c14cdb6ec8434883e/src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_ReturnKey.cs#L19

With this PR, we just restore the behavior defined in the legacy completion.

Comment (dpoeschl) [reply src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs]: Thanks!

**Pull request merged**